src/validator.spec.ts

```diff
@@ -677,4 +678,63 @@ describe("Validator", () => {
         expect(Object.keys(validationErrors2[0].children[0].children[0].constraints).length).toBe(1);
         expect(Object.keys(validationErrors2[0].children[0].children[0].constraints)[0]).toBe("IS_STRING");
     })
+
+    it("should validate only the properties if the group is specified", async () => {
+        class NestedClass {
+            @IsInt({
+                groups: ["GroupA"]
+            })
+            @IsString({
+                groups: ["GroupB"]
+            })
+            // @ts-ignore
+            title: string;
+
+            @IsString()
+            // @ts-ignore
+            description: string;
+        }
+
+        class BasicClass {
+            @validateNested()
+            // @ts-ignore
+            nestedClass: NestedClass;
+
+            @IsString()
+            @IsOptional({
+                groups: ["GroupA"]
+            })
+            // @ts-ignore
+            title?: string;
+        }
+
+        const basicClass = new BasicClass();
+        const nestedClass = new NestedClass();
+        // @ts-ignore
+        nestedClass.title = 3;
+        nestedClass.description = "Description"
+
+        basicClass.nestedClass = nestedClass;
+
+        const validator = new Validator();
+
+        const validationErrorsGroupA =  await validator.validate(basicClass, {
+            groups: ["GroupA"],
+        });
+
+        expect(validationErrorsGroupA.length).toBe(0);
+
+        const validationErrorsGroupB =  await validator.validate(basicClass, {
```

Comment:
We should add a check where we validate directly the nested one. And we should also add a group on the validateNested.